src/chat_do.py

```
@@ -0,0 +1,406 @@
+def _derive_key(secret: str) -> bytes:
+    import hashlib
+    return hashlib.sha256(secret.encode("utf-8")).digest()
+
+
+def _encrypt_xor(plaintext: str, secret: str) -> str:
+    if not plaintext:
+        return ""
+    key = _derive_key(secret)
+    data = plaintext.encode("utf-8")
+    ks = (key * (len(data) // len(key) + 1))[: len(data)]
+    return base64.b64encode(bytes(a ^ b for a, b in zip(data, ks))).decode("ascii")
```

Ruff (0.16.0)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)